jujugui/static/gui/src/app/components/user-profile/entity/entity.js

```diff
@@ -249,28 +250,33 @@ YUI.add('user-profile-entity', function() {
     },
 
     render: function() {
-      var props = this.props;
-      var entity = props.entity;
-      var type = props.type;
-      var isModel = type === 'model';
-      var isCharm = type === 'charm';
+      const props = this.props;
+      const entity = props.entity;
+      const type = props.type;
+      const isModel = type === 'model';
+      const isCharm = type === 'charm';
       // Model names will be in the format "username/model-name" so we have to
       // extract the part we need.
-      var name = entity.name;
+      const name = entity.name;
       if (isModel && entity.name.indexOf('/') !== -1) {
         name = name.split('/')[1];
       }
-      var id = isModel ? entity.uuid : entity.id;
-      var buttonAction = isModel ? this._switchEnv.bind(
+      const id = isModel ? entity.uuid : entity.id;
+      const buttonAction = isModel ? this._switchEnv.bind(
         this, id, name) : this._viewEntity.bind(this, id);
-      var icon = isCharm ? (
+      const icon = isCharm ? (
         <img className="user-profile__entity-icon"
           src={entity.icon}
           title={name} />) : undefined;
-      var classes = {
+      const classes = {
         'user-profile__entity': true,
         'user-profile__list-row': true
       };
+      const destroyButton = isModel && !entity.isAdmin ? (
```

Comment:
In this component (UserProfileEntity) an "entity" can be a model, charm, or bundle. In this line we first check if it is a model (`isModel`) and then make sure that, if it is a model, that it is not a controller (`entity.isAdmin`). I agree that the "entity" term suffers from overload here, but that's a larger problem outside the scope of this PR.